src/manage_nsfs/nc_master_key_manager.js

```diff
@@ -123,20 +153,35 @@ class NCMasterKeysManager {
         throw new Error('init_from_file exhausted');
     }
 
+
+    /**
+     * get_master_keys_json return master keys as JSON object
+     * @param {MasterKey} new_master_key
+     * @returns {string}
+     */
+    _get_master_keys_json(new_master_key) {
+        return JSON.stringify({ active_master_key: new_master_key });
```

Comment:
1. I would add a small comment saying that this is currently limited to one key, but will be changed to support previous inactive keys for rotation.
2. I would add a timestamp inside that json just in case so we could track when it was updated.
3. Do you prefer to write the master_keys_by_id as is and then [ACTIVE_MASTER_KEY] will only contain the id of the active one?
4. Use the const key name -

```suggestion
        return JSON.stringify({ [ACTIVE_MASTER_KEY]: new_master_key });
```

Reply:
1 + 4 - np, changed.
Regarding the other topics - 
2. id contains timestamp, is that enough, or you mean other id?
3. make sense, it might take me a while to do these changes

Reply:
alright, added timetsamp as well